File: leetCode/fukun/201-300/lc_235.js
var lowestCommonAncestor = function (root, p, q) {
    let left = Math.min(p.val, q.val);
    let right = Math.max(p.val, q.val);
    const helper = function (node) {
        if (!node) return false;
        if (node.val < left) {
            return helper(node.right);
        }
        if (node.val > right) {
            return helper(node.left);
        }
        return node;
    }
    return helper(root);
};

File: leetCode/XSS/lc_235.java
public class lc_235 {
    public static void main(String[] args) {
        //二叉搜索数的最近公共祖先
    }
    static class Solution {
        public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
            if(p.val<root.val&&q.val<root.val)
            {
                return lowestCommonAncestor(root.left,p,q);
            }
            else if(p.val>root.val&&q.val>root.val)
            {
                return lowestCommonAncestor(root.right,p,q);
            }
            else return root;
        }
    }
}
